src/pe_reports/report_generator.py

```diff
@@ -20,22 +43,74 @@ def load_template():
     return prs
 
 
-def export_set(prs):
+def export_set(prs, out_dir):
     """Export PowerPoint report set to output directory."""
-    pptx_out = "Customer_ID_Posture_Exposure.pptx"
-    prs.save(os.path.join(SAVE_PATH, pptx_out))
+    try:
+        prs.save(os.path.join(out_dir, REPORT_OUT))
+    except FileNotFoundError as not_found:
+        logging.error("%s : Missing input data. No report generated.", not_found)
     return
 
 
+def generate_reports(data, data_dir, out_dir):
+    """Gather assets to produce reports."""
+    # TODO: build code to connect customer db, encrypt and embed pdf reports.
+    # Issue #7: https://github.com/cisagov/pe-reports/issues/7
+
+
 def main():
+    """Set up logging and call the generate_reports function."""
+    args: Dict[str, str] = docopt.docopt(__doc__, version=__version__)
+    # Validate and convert arguments as needed
+    schema: Schema = Schema(
+        {
+            "--log-level": And(
+                str,
+                Use(str.lower),
+                lambda n: n in ("debug", "info", "warning", "error", "critical"),
+                error="Possible values for --log-level are "
+                + "debug, info, warning, error, and critical.",
+            ),
+            str: object,  # Don't care about other keys, if any
+        }
+    )
+
+    try:
+        validated_args: Dict[str, Any] = schema.validate(args)
+    except SchemaError as err:
+        # Exit because one or more of the arguments were invalid
+        print(err, file=sys.stderr)
+        return 1
+
+    # Assign validated arguments to variables
+    log_level: str = validated_args["--log-level"]
+
+    # Set up logging
+    logging.basicConfig(
+        format="%(asctime)-15s %(levelname)s %(message)s", level=log_level.upper()
+    )
+
+    # TODO: Add generate_reports func to handle cmd line arguments and function.
+    # Issue #8: https://github.com/cisagov/pe-reports/issues/8
+    generate_reports(
+        validated_args["REPORT_DATE"],
+        validated_args["DATA_DIRECTORY"],
+        validated_args["OUTPUT_DIRECTORY"],
+    )
+
     """Generate PDF reports."""
-    print("\n [Info] Loading Posture & Exposure Report Template, Version:", __version__)
+    logging.info(f"Loading Posture & Exposure Report Template, Version : {__version__}")
```

Comment:
Do not use f-strings in calls to the `logging` library.

```suggestion
    logging.info("Loading Posture & Exposure Report Template, Version : %s", __version__)
```

Reply:
Resolved in 4088918